### PR TITLE
[OKX] Xlayer testnet Integration

### LINKED ISSUE #1537: Test integration of Human Protocol with OKTC testnet
**Description**
Test integration of Human Protocol with OKTC testnet

**Motivation**
Test OKTC usability for Human Protocol 

**Implementation details** 
1. Deploy contracts
2. Add OKTC testnet to Job Launcher and Reputation Oracle
3. Test to launch jobs using Job Launcher
4. Test payouts in OKTC using Reputation Oracle
5. Subgraph and dashboard should reflect the integration

### PR DESCRIPTION
**Pull request by mrhouzlane**
## Description

Integration of Human Protocol with X1 testnet : 
- Launch jobs on X1 Testnet 
- Subgraph and Dashboard reflecting the integration. 

## Summary of changes

- Add X1 testnet as custom chain to deploy contracts. 
- To get tokens : [Faucet](https://www.okx.com/x1/faucet) 

## How test the changes

- yarn install inside packages/core
- Fill out .env variables for OKT 
- yarn deploy --network x1testnet
- check address on okt chain block explorer :https://www.oklink.com/fr/x1-test 

### Requirement before launching Jobs : 

Follow these [requirements](https://tech-docs.humanprotocol.org/architecture/job-launcher/installation-and-setup) for X1 Testnet. 

#### Addresses on X1 Testnet : 
[HMT Address](https://www.oklink.com/fr/x1-test/address/0x28838a007bd1cf731fe0ca41d07e1fd02ab133e8) 
[Staking Proxy](https://www.oklink.com/fr/x1-test/address/0x703936a24dfadcd34008db338fdb087c43c8c56a)
[Staking Implementation](https://www.oklink.com/fr/x1-test/address/0x5a71e92c623a6b961b5ff3eb7e9d21c3addcd9a2)
[Escrow Factory Proxy](https://www.oklink.com/fr/x1-test/address/0x83bf8975f4b1b99969b5134f86e5bd683dcf5afc)
[Escrow Factory Implementation](https://www.oklink.com/fr/x1-test/address/0x15374bddd408429096fab19dfe4685de8bb0445c)
[KVStore](https://www.oklink.com/fr/x1-test/address/0xccff9646abf988d9842d94b171a67d5f6c30dbe2)
[Reward Pool Proxy](https://www.oklink.com/fr/x1-test/address/0x44fad6e8d616ad460cf8eef503fc101170e7de73)
[Reward Pool Implementation](https://www.oklink.com/fr/x1-test/address/0x7ea1ff554851c1b381a5f2ed6ea4d72bf6fc40c2)

**Query URL** : https://www.okx.com/api/v1/x1-testnet/index/subgraphs/name/human-protocol-x1-testnet
**Query Playground**  : https://www.okx.com/api/v1/x1-testnet/index/subgraphs/name/human-protocol-x1-testnet/graphql

1. Interaction with contracts  👍 

Scanner :  [X1 testnet](https://www.oklink.com/fr/x1-test).
Contract addresses [here](https://github.com/contracts/contract-addresses).
Updated contracts addresses [here](https://github.com/humanprotocol/human-protocol/blob/main/CONTRACTS_LIST.md).

## Related issues
Closes #1537

### CODE REVIEW

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/job-launcher/client/src/constants/chains.ts [hunk 1]
```diff
@@ -4,20 +4,29 @@ export const IS_MAINNET =
...
+switch (import.meta.env.VITE_APP_ENVIRONMENT.toLowerCase()) {
+  case 'mainnet':
+    SUPPORTED_CHAIN_IDS = [ChainId.POLYGON];
+    break;
+  case 'testnet':
+    SUPPORTED_CHAIN_IDS = [
       ChainId.BSC_TESTNET,
       ChainId.POLYGON_MUMBAI,
       ChainId.GOERLI,
-      ChainId.X1_TESTNET,
```
Comment (portuu3) [inline packages/apps/job-launcher/client/src/constants/chains.ts]: You are removing what you added in this PR

File: packages/apps/job-launcher/client/src/constants/chains.ts [hunk 2]
```diff
@@ -4,20 +4,29 @@ export const IS_MAINNET =
...
+  default:
+    SUPPORTED_CHAIN_IDS = [ChainId.LOCALHOST];
+    break;
+}
 
 export const CHAIN_ID_BY_NAME: Record<string, number> = {
   'Polygon Mumbai': ChainId.POLYGON_MUMBAI,
   'Binance Smart Chain': ChainId.BSC_MAINNET,
   'Ethereum Goerli': ChainId.GOERLI,
-  'X1 Testnet': ChainId.X1_TESTNET,
```
Comment (portuu3) [inline packages/apps/job-launcher/client/src/constants/chains.ts]: You are removing what you added in this PR

**Review by portuu3 — CHANGES_REQUESTED** [review]

File: packages/apps/job-launcher/client/src/constants/chains.ts (hunk 1 above)
Comment (mrhouzlane) [reply packages/apps/job-launcher/client/src/constants/chains.ts]: have undo this

File: packages/apps/job-launcher/client/src/constants/chains.ts (hunk 2 above)
Comment (mrhouzlane) [reply packages/apps/job-launcher/client/src/constants/chains.ts]: have undo this

File: .github/workflows/cd-subgraph.yaml [hunk 3]
```diff
@@ -40,6 +40,8 @@ jobs:
             graph: celo
           - name: celo-alfajores
             graph: celo-alfajores
+          - name : x1testnet
```
[the remark this answers is not in the document]
Comment (mrhouzlane) [reply .github/workflows/cd-subgraph.yaml:43]: waiting for an answer for this one

**Review by portuu3 — APPROVED** [review]

**Pull request merged**